File: binarygap.cpp
#include <iostream>
using namespace std;
int binaryGap(int n)
{
    int lastone = 40;
    int ans = 0;
    for (int i = 0; i < 40; i++)
    {
        if (n & (1 << i))
        {
            ans = max(ans, i - lastone);
            lastone = i;
        }
    }
    return ans;
}
int main()
{
    int n;
    cin>>n;
    binaryGap(n);
    return 0;
}

File: binary gap.java
class BinaryGap {
    public static void main (String args[])
    {
        BinaryGap binaryGap = new BinaryGap();
        int n = 261;
        System.out.println("Max count : "+binaryGap.binaryValue(n));
    }
    
    public int binaryValue(int N) {
        String binaryValue = Integer.toBinaryString(N);
        System.out.println("binaryValue : "+binaryValue);
        int maxZeroCount = 0;
        int zeroCount = 0;
        for (int i = 0, n = binaryValue.length(); i < n; i++) {
            if(binaryValue.charAt(i) == '0') {
                ++zeroCount;
            } else {
                if(zeroCount > maxZeroCount) {
                    maxZeroCount = zeroCount;
                }
                zeroCount = 0;
            }
        }
        return maxZeroCount;
    }
}